v5/patch.go

```diff
@@ -745,17 +1123,17 @@ func (p Patch) ApplyIndent(doc []byte, indent string) ([]byte, error) {
 	for _, op := range p {
 		switch op.Kind() {
 		case "add":
-			err = p.add(&pd, op)
+			err = p.add(&pd, op, options)
```

Comment:
I really do not like switch cases. Is there a way to apply strategy pattern here?

Reply:
1. I mostly ported an earlier commit. 2. The library seems to use switch for such cases (pun intended). I wouldn't worry too much about it. The goal is to be consistent with the original library and avoid unnecessary changes.